src/coreclr/jit/importer.cpp

```diff
@@ -5425,6 +5425,24 @@ GenTree* Compiler::impCastClassOrIsInstToTree(
     // Optimistically assume the jit should expand this as an inline test
     bool shouldExpandInline = true;
     bool isClassExact       = impIsClassExact(pResolvedToken->hClass);
+    bool isSharedInst       = info.compCompHnd->getClassAttribs(pResolvedToken->hClass) & CORINFO_FLG_SHAREDINST;
```

Comment:
We can avoid the JIT/EE interface call for `!isClassExact` case by inlining the JIT/EE interface into the condition below.